Add READMEs to the session-lock crates

pentamassiv opened the pull request:
Add READMEs to the session-lock crates and update the README in the root of the repo to mention the newly added crates

pentamassiv commented:
@happenslol What do you think of these READMEs? They are pretty much copy-pasted from gtk4-layer-shell.

happenslol commented:
> @happenslol What do you think of these READMEs? They are pretty much copy-pasted from gtk4-layer-shell.

Ah, you were faster than me! Had a few busy days so I didn't get to it. Hope your exams went well!

Looks good to me. The only other thing that might be worth mentioning is the fact that they both require the `.so` file for gtk4-layer-shell, and will work fine together if both crates are used, but **only if the same shared library is imported**.

I'm pretty sure 2 different library versions would overwrite each other's hooks/proxies and cause one of the crates to stop working.

pentamassiv commented:
I added that information to the Readme, but then I changed my mind and restructured all the Readmes. I didn't like the duplication of the description of how to generate the code. I also don't think that information is relevant to most users. It's better to describe what the crate actually does. What's your opinion on it?

happenslol commented:
I wasn't a huge fan of the duplication in the readmes either, so this is great!

Honestly, whenever I cloned the repo to test something locally, I would end up going through the different readmes in order every time, because the workflow is always build sys -> build API, so you start with the root readme, remember you have to go to the sys readme, and then the API readme.

This removes at least one step from that flow 😋
